run/train.py

```diff
@@ -135,11 +140,14 @@ def main(config):
             workspace.save_checkpoint(epoch, log)
 
             print("epochs {}".format(epoch))
+            print("learning rate: {}".format(lr_scheduler.get_lr()))
             print("\ttrain_loss: {}".format(train_results["loss"]))
             print("\ttrain_metric: {}".format(train_results["metric"]))
             print("\tdev_loss: {}".format(dev_results["loss"]))
             print("\tdev_metric: {}".format(dev_results["metric"]))
 
+        lr_scheduler.step()
```

Comment:
I think some schedulers require stepping for every optimization step, e.g., the linear ramp-up then exponential decay scheduler.

Reply:
but if we do so, the scheduler step depends on the size of the dataset and batch size.
Which might confuse the user on what they need to put for the scheduler step config.

I am not saying I don't want to step for every optimization step. I want to discuss so that I can have the best implementation.

Reply:
Okay, maybe have an option that determines the interval of the scheduler stepping, i.e., per-epoch vs. per-optimizer-step.

Reply:
I see, good idea. I will have a separate PR for that